src/sst/elements/osseous/rtlcomponent.h

```diff
@@ -0,0 +1,161 @@
+#include "rtlwriteev.h"
+#include "rtlmemmgr.h"
+#include "AXI_port.h"
+
+namespace SST {
+    namespace RtlComponent {
+
+class Rtlmodel : public SST::Component {
+
+public:
+	Rtlmodel( SST::ComponentId_t id, SST::Params& params );
+    ~Rtlmodel();
+
+	void setup();
+	void init(unsigned);
+	void finish();
+
+	bool clockTick( SST::Cycle_t currentCycle );
+
+	SST_ELI_REGISTER_COMPONENT(
+		Rtlmodel,
+		"rtlcomponent",
+		"Rtlmodel",
+		SST_ELI_ELEMENT_VERSION( 1, 0, 0 ),
+		"Demonstration of an External Element for SST",
+		COMPONENT_CATEGORY_PROCESSOR
+	)
+    //Stats needs to be added. Now, stats will be added based on the outputs as mentioned by the user based on the RTL login provided. 
+    SST_ELI_DOCUMENT_STATISTICS(
+        { "read_requests",        "Statistic counts number of read requests", "requests", 1},   // Name, Desc, Enable Level
+        { "write_requests",       "Statistic counts number of write requests", "requests", 1},
+        { "read_request_sizes",   "Statistic for size of read requests", "bytes", 1},   // Name, Desc, Enable Level
+        { "write_request_sizes",  "Statistic for size of write requests", "bytes", 1},
+        { "split_read_requests",  "Statistic counts number of split read requests (requests which come from multiple lines)", "requests", 1},
+        { "split_write_requests", "Statistic counts number of split write requests (requests which are split over multiple lines)", "requests", 1},
+	    { "flush_requests",       "Statistic counts instructions which perform flushes", "requests", 1},
+        { "fence_requests",       "Statistic counts instructions which perform fences", "requests", 1}
+    )
+    //Parameters will mostly be just frequency/clock in the design. User will mention specifically if there could be other parameters for the RTL design which needs to be configured before runtime.  Don't mix RTL input/control signals with SST parameters. SST parameters of RTL design will make the RTL design/C++ model synchronous with rest of the SST full system.   
+	SST_ELI_DOCUMENT_PARAMS(
+		{ "ExecFreq", "Clock frequency of RTL design in GHz", "1GHz" },
+		{ "maxCycles", "Number of Clock ticks the simulation must atleast execute before halting", "1000" },
+        {"memoryinterface", "Interface to memory", "memHierarchy.memInterface"}
+	)
+
+    //Default will be single port for communicating with Ariel CPU. Need to see the requirement/use-case of multi-port design and how to incorporate it in our parser tool.  
+    SST_ELI_DOCUMENT_PORTS(
+        {"ArielRtllink", "Link to the Rtlmodel", { "Rtlmodel.RTLEvent", "" } },
+        {"RtlCacheLink", "Link to Cache", {"memHierarchy.memInterface" , ""} }
+    )
+    
+    SST_ELI_DOCUMENT_SUBCOMPONENT_SLOTS(
+            {"memmgr", "Memory manager to translate virtual addresses to physical, handle malloc/free, etc.", "SST::RtlComponent::RtlMemoryManager"},
+            {"memory", "Interface to the memoryHierarchy (e.g., caches)", "SST::Interfaces::SimpleMem" }
```

Comment:
This is true in a lot of these, can you easily switch this over to StandardMem instead? Otherwise, when I do the switchover of Ariel, we can circle back and update this.

Reply:
Replaced SimpleMem to StandardMem in 84416d6